Sources/LocalReceiptParsing/LocalReceiptFetcher.swift

```diff
@@ -7,13 +7,17 @@
 //
 //      https://opensource.org/licenses/MIT
 //
-//  ReceiptFetcher.swift
+//  LocalReceiptFetcher.swift
 //
-//  Created by Nacho Soto on 1/10/23.
+//  Created by Mark Villacampa on 8/1/24.
 
 import Foundation
 
-public extension PurchasesReceiptParser {
+internal protocol LocalReceiptFetcherType: Sendable {
+    func fetchAndParseLocalReceipt() throws -> AppleReceipt
+}
+
+internal final class LocalReceiptFetcher: LocalReceiptFetcherType {
```

Comment:
`ReceiptFetcher` depends on `StoreKitRequestFetcher` and `SystemInfo` (which would create a circular dependency BundleSandboxEnvironmentDetector).

So instead of adding these methods as an extension to `PurchasesReceiptParser`, I've created `LocalReceiptFetcher` as a simplified version of `ReceiptFetcher`, with an accompanying protocol so it can be tested.

Ideally we would be able to use `ReceiptFetcher` but the dependencies make it hard :/ any other suggestion to simplify this?

Reply:
Yeah that's why we had this. In the future we'll be able to share code using the `package` visibility.